Sources/Snowplow/Ecommerce/Events/CheckoutStepEvent.swift

```diff
@@ -0,0 +1,142 @@
+public class CheckoutStepEvent: SelfDescribingAbstract {
+    /// Checkout step index.
+    @objc
+    public var step: Int
+    
+    /// Shipping address postcode.
+    @objc
+    public var shippingPostcode: String?
+    
+    /// Billing address postcode.
+    @objc
+    public var billingPostcode: String?
+    
+    /// Full shipping address.
+    @objc
+    public var shippingFullAddress: String?
+    
+    /// Full billing address.
+    @objc
+    public var billingFullAddress: String?
+    
+    /// Can be used to discern delivery providers DHL, PostNL etc.
+    @objc
+    public var deliveryProvider: String?
+    
+    /// E.g. store pickup, standard delivery, express delivery, international.
+    @objc
+    public var deliveryMethod: String?
+    
+    /// Coupon applied at checkout.
+    @objc
+    public var couponCode: String?
+    
+    /// Type of account used on checkout, e.g. existing user, guest.
+    @objc
+    public var accountType: String?
+    
+    /// Any kind of payment method the user selected to proceed. Card, PayPal, Alipay etc.
+    @objc
+    public var paymentMethod: String?
+    
+    /// E.g. invoice or receipt
+    @objc
+    public var proofOfPayment: String?
+    
+    /// If opted in to marketing campaigns to the email address.
+    public var marketingOptIn: Bool?
+    
+    override var schema: String {
+        return ecommerceActionSchema
+    }
+    
+    override var payload: [String : Any] {
+        let data: [String: Any] = ["type": "checkout_step"]
+        return data
+    }
+    
+    override internal var entitiesForProcessing: [SelfDescribingJson]? {
+        get {
+            var data: [String : Any] = ["step": step]
+            
+            if let shippingPostcode = shippingPostcode { data["shipping_postcode"] = shippingPostcode }
+            if let billingPostcode = billingPostcode { data["billing_postcode"] = billingPostcode }
+            if let shippingFullAddress = shippingFullAddress { data["shipping_full_address"] = shippingFullAddress }
+            if let billingFullAddress = billingFullAddress { data["billing_full_address"] = billingFullAddress }
+            if let deliveryProvider = deliveryProvider { data["delivery_provider"] = deliveryProvider }
+            if let deliveryMethod = deliveryMethod { data["delivery_method"] = deliveryMethod }
+            if let couponCode = couponCode { data["coupon_code"] = couponCode }
+            if let accountType = accountType { data["account_type"] = accountType }
+            if let paymentMethod = paymentMethod { data["payment_method"] = paymentMethod }
+            if let proofOfPayment = proofOfPayment { data["proof_of_payment"] = proofOfPayment }
+            if let marketingOptIn = marketingOptIn { data["marketing_opt_in"] = marketingOptIn }
+            
+            return [SelfDescribingJson(schema: ecommerceCheckoutStepSchema, andData: data)]
+        }
+    }
+    
+    /// - Parameter step Checkout step index.
```

Comment:
I'm not sure if this will be picked up by the docs generator without the semicolon, just to be sure, I'd add it everywhere.

```suggestion
    /// - Parameter step: Checkout step index.
```